### PR TITLE
Verify machine-api components are functional after upgrade

### PR DESCRIPTION
**Pull request by jhou1**
As subject, I'd like to to have CI automatically cover upgrade test scenarios. I've tested them locally. @geliu2016 @pruan-rht @akostadinov Would you please review? 

```
4 scenarios (4 passed)
31 steps (31 passed)
5m59.788s
[08:29:37] INFO> === At Exit ===
```

### CODE REVIEW

**Comment by geliu2016** [comment]
LGTM, you may try it on runner-v3 also.

File: features/upgrade/machines/machines.feature [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    And I pick a random machineset to scale
+    And evaluation of `machine_set.available_replicas` is stored in the :replicas_to_restore clipboard
+
+    Given I scale the machineset to +2
+    And I register clean-up steps:
+    """
+    When I scale the machineset to <%= cb.replicas_to_restore %>
+    Then the machineset should have expected number of running machines
+    """
+    Then the step should succeed
```
Comment (akostadinov) [inline features/upgrade/machines/machines.feature]: if scaling operation fails, do we need to restore? Perhaps put clean-up step after this step?
Or can scale operation succeed but command to fail?

Comment (akostadinov) [reply features/upgrade/machines/machines.feature]: other than that LGTM

Comment (jhou1) [reply features/upgrade/machines/machines.feature]: I'll put it after this step, thank you.

File: features/upgrade/machines/machines.feature [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    Given evaluation of `cluster_operator(<cluster_operator>).condition(type: 'Progressing')` is stored in the :co_progressing clipboard
+    Then the expression should be true> cb.co_progressing["status"]=="False"
+
+  Examples:
+    | cluster_operator     |
+    | "machine-api"        | # @case_id OCP-22712
+    | "cluster-autoscaler" | # @case_id OCP-27664
+
+  # @author jhou@redhat.com
+  # @case_id
```
Comment (liangxia) [inline features/upgrade/machines/machines.feature]: No ID ?

**Pull request merged**